### PR TITLE
:sparkles: feat(components): define component a-modal

### LINKED ISSUE #39: Web: M - Modal
Create a new Modal component.

Examples usage:

```tsx
<Modal open={true} onClose={() => console.log('Close')}>Hello world</Modal>
```

## API

| Param    | Description                                             | Type         | Default value |
| -------- | ------------------------------------------------------- | ------------ | ------------- |
| children | Content of component                                    | `ReactNode`  | -             |
| open     | If `true`, the component is shown                       | `boolean`    | -             |
| onClose  | Callback fired when the component requests to be closed | `() => void` | -             |

### PR DESCRIPTION
**Pull request by dinhthanhtai**
Closes #39

### CODE REVIEW

**Comment by thaihoanganh** [comment]
@dinhthanhtai #39 issue name here `Web: M - Modal` does that mean this component belongs to molecules, `M` abbreviation is molecules.

File: apps/web/src/components/atoms/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import classNames from 'classnames';
+import { createPortal } from 'react-dom';
+
+import './styles.scss';
+
+export interface ModalProps {
+  open?: boolean;
+  onClose?: () => void;
+  className: object;
+  children: React.ReactNode;
```
Comment (thaihoanganh) [inline apps/web/src/components/atoms/Modal/Modal.tsx:12]: You using `React.FC<ModalProps>` for the type of this component, you should understand the type of `React.FC` 
already exist `children` with type is `React.ReactNode`. Remove `children` if you do not overwrite `children` of `React.FC`.

File: apps/web/src/components/atoms/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export const Modal: FC<ModalProps> = ({ className, children, open = false, onClose, ...otherProps }) => {
+  const modal = (
+    <div
+      className={classNames('a-modal', {
+        className,
+      })}
+      {...otherProps}
+      onClick={onClose}
```
Comment (thaihoanganh) [inline apps/web/src/components/atoms/Modal/Modal.tsx:22]: Something is wrong. Here, when Modal is open you can close Modal by clicking anywhere. You should close Modal when to click outside the content of a modal.

**Comment by dinhthanhtai** [comment]
Could you help me delete this branch @thaihoanganh?. I have just made a new one [here ](https://github.com/VC-Team/yemusic/pull/67)

**Comment by thaihoanganh** [comment]
Closes, this PR has moved to a new PR in [here](https://github.com/VC-Team/yemusic/pull/67)

**Pull request closed without merging**